bootstrap/v2/pkg/apis/apps/kfdef/v1alpha1/application_types.go

```diff
@@ -655,3 +681,101 @@ func IsSecretNotFound(e error) bool {
 	_, ok := e.(*SecretNotFound)
 	return ok
 }
+
+func getParameter(parameters []config.NameValue, paramName string) (string, bool) {
+	for _, p := range parameters {
+		if p.Name == paramName {
+			return p.Value, true
+		}
+	}
+
+	return "", false
+}
+
+func setParameter(parameters []config.NameValue, paramName string, value string) []config.NameValue {
```

Comment:
should we do it in place (modify the input directly)? So that we don't have to use the return value
 a = setParameter(a, xx)

Reply:
@lluunn because of how slices work in Go, it is necessary to return the resulting slice.
[Excerpt from golang.org](https://blog.golang.org/slices):

> It's important to understand that even though a slice contains a pointer, it is itself a value. Under the covers, it is a struct value holding a pointer and a length. It is not a pointer to a struct. 

In other words, if you have appended the slice in a function and then don't return it, the slice of the calling function may contain obsolete data.
[relevant stackoverflow issue](https://stackoverflow.com/questions/20195296/golang-append-an-item-to-a-slice)

Reply:
@yanniszark is correct. I think  @lluunn might be asking why not take (*KfDef).

This method is specifically intended to be a helper method called from SetApplicationParameter.
So we refactored out the common bits.  I didn't want to make it a func (d*KfDef) function because that makes it less clear what the function is doing; i.e. the function could be reading modifying any (d *kfDef). The current function signature is more narrowly scoped which I think is better.

Also its beneficial in that the function def can have fewer side effects (i.e. modifying *KfDef).